File: ios/IosLocalNetworkPermission.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(IosLocalNetworkPermission, NSObject)

RCT_EXTERN_METHOD(check: (RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

@end

File: ios/IosLocalNetworkPermission.swift
@objc(IosLocalNetworkPermission)
class IosLocalNetworkPermission: NSObject {

  @objc
  func check(_ resolve: @escaping RCTPromiseResolveBlock,reject: @escaping RCTPromiseRejectBlock) -> Void {
   self.checkLanAccess{isGranted in 
    if(isGranted) {
      resolve("true")
    } else {
      resolve("false")
    }
   }
  }

  func checkLanAccess(_ completed: Optional<(Bool) -> Void> = .none) {
    DispatchQueue.global(qos: .userInitiated).async {
      let hostName = Process.processInfo.hostName
      let isGranted = hostName.contains(".local")
      if let completed {
        DispatchQueue.main.async {
          completed(isGranted)
        }
      }
    }
  }
}
